pkg/intfmapper/device.go

```diff
@@ -85,10 +88,30 @@ func (d *device) collector() {
 }
 
 func (d *device) collect() error {
-	s := *gosnmp.Default
-	s.Community = d.community
-	s.Target = d.addr.String()
-	s.Timeout = time.Second * 30
+	s := &gosnmp.GoSNMP{
+		Target:                  d.addr.String(),
+		Port:                    snmpPort,
+		Community:               d.snmpCfg.Community,
+		Version:                 gosnmp.Version2c,
+		Timeout:                 timeout,
+		Retries:                 0,
+		ExponentialTimeout:      false,
+		UseUnconnectedUDPSocket: true,
+	}
+
+	if d.snmpCfg.Version == 3 {
+		s.Community = ""
+		s.Version = gosnmp.Version3
+		s.SecurityModel = gosnmp.UserSecurityModel
+		s.MsgFlags = gosnmp.AuthPriv
+		s.SecurityParameters = &gosnmp.UsmSecurityParameters{
+			UserName:                 d.snmpCfg.User,
+			AuthenticationProtocol:   gosnmp.SHA,
```

Comment:
Might make sense to also make `MsgFlags`, `AuthenticationProtocol`, `PrivacyProtocol` configurable as well? These values should work for most folks though :-D